### PR TITLE
fix: fix search function to accept null criteria as empty query string

### LINKED ISSUE #1137: TypeError: App\Http\ViewHelpers\Adminland\AdminExpenseViewHelper::search(): Argument #2 ($criteria) must be ...
Sentry Issue: [FORTRABBIT-Z](https://sentry.io/organizations/officelife/issues/2475172534/?referrer=github_integration)

```
TypeError: App\Http\ViewHelpers\Adminland\AdminExpenseViewHelper::search(): Argument #2 ($criteria) must be of type string, null given, called in /srv/app/officelifehq/htdocs/app/Http/Controllers/Company/Adminland/AdminExpenseController.php on line 136
  File "/app/Http/ViewHelpers/Adminland/AdminExpenseViewHelper.php", line 68, in App\Http\ViewHelpers\Adminland\AdminExpenseViewHelper::search
    public static function search(Company $company, string $criteria): Collection
  File "/app/Http/Controllers/Company/Adminland/AdminExpenseController.php", line 136, in App\Http\Controllers\Company\Adminland\AdminExpenseController::search
    $employees = AdminExpenseViewHelper::search($loggedCompany, $request->input('searchTerm'));
  File "/app/Http/Middleware/CheckHRRole.php", line 24, in App\Http\Middleware\CheckHRRole::handle
    return $next($request);
  File "/app/Http/Middleware/CheckCompany.php", line 44, in App\Http\Middleware\CheckCompany::handle
    return $next($request);
  File "/app/Http/Middleware/EnsureEmailIsVerified.php", line 21, in App\Http\Middleware\EnsureEmailIsVerified::handle
    return parent::handle($request, $next, $redirectToRoute);
...
(66 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by asbiin**
Fix #1137 

You fool, don't forget these steps:

- [ ] Unit tests
- [ ] Tests with Cypress
- [ ] Documentation
- [ ] Dummy data

### CODE REVIEW

File: app/Http/ViewHelpers/Adminland/AdminSoftwareViewHelper.php [hunk 1]
```diff
@@ -132,16 +132,15 @@ public static function seats($employees, Company $company): Collection
...
      */
-    public static function getPotentialEmployees(Software $software, Company $company, string $criteria): Collection
+    public static function getPotentialEmployees(Software $software, Company $company, ?string $criteria): Collection
     {
         // get list of employees who have the software
         $employees = $software->employees()
             ->select('id')
-            ->pluck('id')
-            ->toArray();
+            ->pluck('id');
```
Comment (djaiss) [inline app/Http/ViewHelpers/Adminland/AdminSoftwareViewHelper.php:143]: Are you sure? `pluck` returns a collection, not an array.

File: app/Http/ViewHelpers/Company/HR/CompanyHRViewHelper.php [hunk 2]
```diff
@@ -30,9 +30,7 @@ public static function eCoffees(Company $company): ?array
         $allMatchesInCompany = DB::table('e_coffees')
             ->where('company_id', $company->id)
             ->select('id')
-            ->get()
-            ->pluck('id')
-            ->toArray();
+            ->pluck('id');
```
Comment (djaiss) [inline app/Http/ViewHelpers/Company/HR/CompanyHRViewHelper.php:33]: Same comment as above. Pluck returns a collection.

File: app/Http/ViewHelpers/Employee/EmployeeHierarchyViewHelper.php [hunk 3]
```diff
@@ -27,26 +35,15 @@ public static function search(Company $company, Employee $employee, string $crit
                     ->orWhere('email', 'LIKE', '%'.$criteria.'%');
             })
             ->where('id', '!=', $employee->id)
+            ->whereNotIn('id', $existingManagersForTheEmployee->merge($existingDirectReportsForTheEmployee))
```
Comment (djaiss) [inline app/Http/ViewHelpers/Employee/EmployeeHierarchyViewHelper.php:38]: Even after reading the doc, I don't understand what `merge` does here.

**Review by djaiss — COMMENTED** [review]
Overall very good changes, but there are two-three changes I don't understand.

File: app/Http/ViewHelpers/Adminland/AdminSoftwareViewHelper.php (hunk 1 above)
Comment (asbiin) [reply app/Http/ViewHelpers/Adminland/AdminSoftwareViewHelper.php:143]: Laravel will convert it to an array when required.

File: app/Http/ViewHelpers/Company/HR/CompanyHRViewHelper.php (hunk 2 above)
Comment (asbiin) [reply app/Http/ViewHelpers/Company/HR/CompanyHRViewHelper.php:33]: Laravel will convert it to an array when required.

File: app/Http/ViewHelpers/Employee/EmployeeHierarchyViewHelper.php (hunk 3 above)
Comment (asbiin) [reply app/Http/ViewHelpers/Employee/EmployeeHierarchyViewHelper.php:38]: both collection contains a list of ids.
`merge` will merge the 2 collections.

**Pull request merged**